src/factory.js

```diff
@@ -32,81 +33,196 @@ import WhichDelete from './commands/which-delete'
 import WhichReport from './commands/which-report'
 import WhichStatus from './commands/which-status'
 import WhichMap from './commands/which-map'
+import BroadcastService from './services/broadcast-service'
+import RepeatService from './services/repeat-service'
+import SendService from './services/send-service'
+import FileService from './services/file-service'
+import GenericService from './services/generic-service'
+import APIService from './services/api-service'
+import StatusService from './services/status-service'
+import ReportService from './services/report-service'
+import Version from './commands/version'
 
 // TODO how can we use a new Broadcast service each time???
 class Factory {
   // TODO add send service
-  constructor(
-    broadcastService,
-    sendService,
-    statusService,
-    repeatService,
-    reportService,
-    genericService,
-    fileService
-  ) {
+  constructor({ messageService }) {
+    // constructor({ messageService, broadcastService }) {
     // These are the services that will be passed to the commands
-    this.broadcastService = broadcastService
-    this.sendService = sendService
-    this.statusService = statusService
-    this.repeatService = repeatService
-    this.reportService = reportService
-    this.genericService = genericService
-    this.fileService = fileService
+    this.messageService = messageService
+    // this.broadcastService = broadcastService
+    this.apiService = new APIService()
+    this.reportService = new ReportService()
+
+    const userState = messageService.getUserCurrentState({
+      userEmail: this.messageService.userEmail,
+    })
+    console.log({ userState })
+    this.broadcastService = new BroadcastService({
```

Comment:
The services should be passed in so it will be easier to change them in the future

Reply:
you can change them in the factory? Not sure i understand the benefit of moving them out and passing them to the factory. Id rather put a function that selects whatever service inside the factory, and then call it and assign the result inside the constructor,  instead of calling the function in main, and passing the result to the factory